src/users/UserList.js

```diff
@@ -56,14 +65,7 @@ export default function UserList(props: Props) {
       sections={sections}
       keyExtractor={item => item}
       getItemLayout={getItemLayout}
-      renderItem={({ item }) => (
-        <UserItem
-          key={item}
-          userId={item}
-          onPress={onPress}
-          isSelected={!!selected.find(user => user.user_id === item)}
-        />
-      )}
+      renderItem={renderItem}
```

Comment:
Ah, yeah, thanks for catching that. I've put it in a useCallback, since it relies on some props. I don't think this'll make a big perf difference, but it seems worth doing.

Reply:
Cool, makes sense. Looking at more of the code, it actually don't expect this to have any effect at all—there are a few other props that may be causing unnecessary rerenders exactly where `renderItem` was:

- `keyExtractor` (which seems a bit belt-and-suspenders anyway, so we may be able to just remove—we're already giving `UserItem` a `key`).
- `renderSectionHeader`

Reply:
Separately, with a handy [setting](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/eslint.3A.20Flow.20Language.20Support/near/1213702) in the Flow extension for VSCode, I see a small type-checking hole like the one we addressed (a bit) in c37e5b9141b0135e3bb1403e5f008d2e1df8f73b.

```diff
- renderItem={({ item }) => (
+ renderItem={({ item }: { item: UserId }) => (
```

When I run `yarn flow` after that, I don't get any error output. 🎉 (Er, actually, there's one error I'd hope it would flag but it doesn't, not sure why: we should have to call `.toString()` on `item` for `UserItem`'s `key` attribute because `key` is [supposed to be a string](https://reactjs.org/docs/lists-and-keys.html).)

Also, poking around a bit in RN, it looks like we could do this, for `renderSectionHeader`:

```diff
+ import type { SectionBase } from 'react-native/Libraries/Lists/VirtualizedSectionList';

[...]

- renderSectionHeader={({ section }) =>
+ renderSectionHeader={({ section }: { section: SectionBase<UserId> }) =>
```

In addition to giving us more type coverage, I think it's a bit nicer to have these signposts to help us understand what's contained in the vaguely-named `item` and `section` objects.